src/lib/features/metrics/instance/metrics.ts

```diff
@@ -150,16 +155,25 @@ export default class ClientMetricsController extends Controller {
         } else {
             try {
                 const { body: data, ip: clientIp, user } = req;
-                data.environment = this.metricsV2.resolveMetricsEnvironment(
-                    user,
-                    data,
-                );
+                const { impactMetrics, ...metricsData } = data;
```

Comment:
would it be possible to do the application logic in the service and keep the controller as dumb as possible?